### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot read property 'add' of undefined in publisherRestrictions

### ISSUE THREAD

**Issue by oztek22** [InteractiveAdvertisingBureau/iabtcf-es#203]
**Version**
1.1.0-1

**Module (core, cmpapi, cli, stub, or testing)**
core

**Describe with reproduction steps – What is the expected behavior?**
After making a change in flexible purposes and trying to add restriction to vendor it gives the following error:

> Uncaught (in promise) TypeError: Cannot read property 'add' of undefined

Code which causes the issue

`
        const purposeRestriction = new PurposeRestriction(purposeId, purposeRestrictionType);
        this.tcModel.publisherRestrictions.add(vendor.id, purposeRestriction);
`

![image](https://user-images.githubusercontent.com/14314089/89775486-85021300-db25-11ea-9210-e3de95e9c96b.png)

**Comment by chrispaterson** [comment]
I'm having trouble replicating this, can you give me some input values that will create this?

**Comment by oztek22** [comment]
Hi @chrispaterson 

following code helps u replicating it
```
        const purposeRestriction = new PurposeRestriction(3, RestrictionType.REQUIRE_CONSENT);
        this.tcModel.publisherRestrictions.add(1, purposeRestriction);
        console.log('passed');
        if (this.tcModel.publisherRestrictions.vendorHasRestriction(1, purposeRestriction)) {
          console.log('restriction checked');
        }
        this.tcModel.publisherRestrictions.add(1, purposeRestriction);
        console.log('again passed');
```
![image](https://user-images.githubusercontent.com/14314089/89975024-49309000-dc82-11ea-90ee-b99a63f93656.png)

By Investigating further we figured that it is deleting the map if the same restriction already exists and that causes the error.
https://github.com/InteractiveAdvertisingBureau/iabtcf-es/blob/master/modules/core/src/model/PurposeRestrictionVector.ts#L308

**Comment by chrispaterson** [comment]
This is fixed in 1.1.0-6
